src/test/fuzz/orphanage.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <net_processing.h>
+#include <primitives/transaction.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/mining.h>
+#include <test/util/script.h>
+#include <test/util/setup_common.h>
+#include <txorphanage.h>
+#include <validation.h>
+
+namespace {
+const TestingSetup* g_setup;
+} // namespace
+
+void initialize_orphanage()
+{
+    static const auto testing_setup = MakeNoLogFileContext<const TestingSetup>();
+    g_setup = testing_setup.get();
+    SyncWithValidationInterfaceQueue();
```

Comment:
I think a basic testing setup (args and logging) is sufficient. No need for a chain here in this simple test.

Reply:
I also want to keep the setup minimal. The chain used here is to construct the first tx by calling `chainman->GetParams().GenesisBlock().vtx[0]`, maybe I should just construct it manually instead of getting it from the genesis block.

Reply:
Yeah, I think it might be preferable to just construct it manually. However, if you switch to an outpoints approach, you'll probably need to create a few outpoints manually instead.

Reply:
Please keep the `MakeNoLogFileContext` in init. It is not strictly needed, but I think it is nice to have to have a logger and argsmanager initialized for testing.

The reason is that they are currently globals, so they may be used at any time during runtime or in future code changes without the fuzz test noticing. And if they are used, we want them to be initialized for testing and not accidentally corrupt the main datadir (or cause OOMs) or other bugs.